src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -617,6 +617,122 @@ public function testDumpDateTime($dateTime, $expected)
         $this->assertSame($expected, Inline::dump($dateTime));
     }
 
+    /**
+     * @dataProvider getNumericKeyData
+     */
+    public function testDumpNumericKeyAsString(array|int $input, int $flags, string $expected)
+    {
+        $this->assertSame($expected, Inline::dump($input, $flags));
+    }
+
+    public function getNumericKeyData()
+    {
+        yield 'Int with flag' => [
+            200,
+            Yaml::DUMP_NUMERIC_KEY_AS_STRING,
+            "'200'",
+        ];
+
+        yield 'Int without flag' => [
+            200,
+            0,
+            '200',
+        ];
+
+        yield 'Int key with flag' => [
+            [200 => 'foo'],
+            Yaml::DUMP_NUMERIC_KEY_AS_STRING,
+            "{ '200': foo }",
+        ];
+
+        yield 'String key with flag' => [
+            ['200' => 'foo'],
+            Yaml::DUMP_NUMERIC_KEY_AS_STRING,
+            "{ '200': foo }",
+        ];
+
+        yield 'Bool key with flag' => [
+            [true => 'foo'],
+            Yaml::DUMP_NUMERIC_KEY_AS_STRING,
+            "{ '1': foo }",
+        ];
```

Comment:
Not sure if we need to test this. PHP does not allow boolean keys. The key is cast to an integer immediately.